var/spack/repos/builtin/packages/qmcpack/package.py

```diff
@@ -270,6 +270,16 @@ def cmake_args(self):
             args.append("-DFFTW_HOME={0}".format(fftw_prefix))
             args.append("-DFFTW_INCLUDE_DIRS={0}".format(fftw_prefix.include))
             args.append("-DFFTW_LIBRARY_DIRS={0}".format(fftw_prefix.lib))
+        elif "^armpl-gcc" in spec:
+            args.append("-DBLA_VENDOR=Arm")
+            args.append("-DBLAS_LIBRARIES={0}".format(spec["armpl-gcc"].libs.joined(";")))
+            args.append("-DFFTW_LIBRARIES={0}".format(spec["armpl-gcc"].libs.joined(";")))
+            args.append("-DFFTW_INCLUDE_DIR={0}".format(spec["armpl-gcc"].headers.directories[0]))
+        elif "^acfl" in spec:
+            args.append("-DBLA_VENDOR=Arm")
+            args.append("-DBLAS_LIBRARIES={0}".format(spec["acfl"].libs.joined(";")))
+            args.append("-DFFTW_LIBRARIES={0}".format(spec["acfl"].libs.joined(";")))
+            args.append("-DFFTW_INCLUDE_DIR={0}".format(spec["acfl"].headers.directories[0]))
```

Comment:
It is not good to have FFT and BLAS control mixed.
Without your change, doesn't ARMPL gets picked up by CMake?
By the way, for BLAS/LAPACK, Arm_mp is preferred.

Reply:
I also see no reason to separate BLAS and FFT libraries settings in this case since one cannot do, e.g.,

spack install qmcpack ^armpl-gcc ^fftw

since both armpl-gcc and fftw provides the virtual fftw-api.

Reply:
I meant, when BLA_VENDOR is not set. CMake picks up the correct version of the libraries, if BLA_VENDOR is set.

In fact, in this commit setting BLA_VENDOR to Arm is redundant. I'd better remove it as it might be confusing. I'll push a new commit. Thanks.

Reply:
It remains unclear to me what spack virtual packages exactly provide (include path, link path...). headerless fortrain libraries may behave one way and C library can be quite different. QMCPACK may allow mixed FFTW if the libraries link interface may distinguish one from another.

Reply:
My initial request remains.
1) please do not mix BLAS and FFTW in the current giant if block
2) Control BLAS using BLA_VENDOR

Reply:
> I also see no reason to separate BLAS and FFT libraries settings in this case since one cannot do, e.g.,
> 
> spack install qmcpack ^armpl-gcc ^fftw
> 
> since both armpl-gcc and fftw provides the virtual fftw-api.

Is this worth adding a `conflicts()` e.g., `conflicts("^armpl-gcc", when="^fftw")`?